server/main_server/src/main.cpp

```diff
@@ -734,6 +737,17 @@ Mandatory arguments to long options are mandatory for short options too.
             return -1;
         }
 
+        // SIGUSR2 is used by the agent factory to notify the main server process
+        // when it has completed initialization.
+        struct sigaction sa_sigusr2; // NOLINT(cppcoreguidelines-pro-type-member-init)
+        sigemptyset(&sa_sigusr2.sa_mask);
+        sa_sigusr2.sa_flags = 0;
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-union-access)
+        sa_sigusr2.sa_handler = [](int) { g_agent_factory_initialized = 1; }; // TODO Should we constrain this to the agent factory?
+        if (sigaction(SIGUSR2, &sa_sigusr2, nullptr) == -1) {
```

Comment:
```suggestion
        if (-1 == sigaction(SIGUSR2, &sa_sigusr2, nullptr)) {
```

Reply:
I'm going to continue with the existing code since the invocation of `sigaction` is not assignable.